### PR TITLE
Add links to the new unit testing doc and update tox info

### PR DESCRIPTION
**Pull request by willbarton**
This PR adds a link to our upcoming [Unittesting Django and Wagtail](https://github.com/cfpb/development/blob/unittesting-django-wagtail/guides/unittesting-django-wagtail.md) documentation and updates the documentation about our tox environments to match current reality. It also removes some of the linting documentation and instead links to the development repo where that info is maintained.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: docs/testing-be.md [hunk 1]
```diff
@@ -1,27 +1,59 @@
...
+## Running tests
+
+To run the the full suite of Python tests using Tox, 
+make sure you are in the cfgov-refresh root and then run:
 
 ```
 tox
 ```
 
+This will run linting tests and unit tests with migrations.
```
Comment (Scotchester) [inline docs/testing-be.md]: This runs the linter by default?

File: docs/testing-be.md [hunk 2]
```diff
@@ -42,13 +74,10 @@ but is not used by defalt when running tests locally.
 ## Source code linting
```
Comment (Scotchester) [inline docs/testing-be.md]: Should this be turned into a subsection of "Running tests"?

File: docs/testing-be.md (hunk 1 above)
Comment (willbarton) [reply docs/testing-be.md]: Yes, it's the same as running `tox -e lint -e unittest-py27-dj111-wag113-slow`.

File: docs/testing-be.md (hunk 2 above)
Comment (willbarton) [reply docs/testing-be.md]: Yeah, it probably can be at this point.

File: docs/testing-be.md [hunk 3]
```diff
@@ -1,95 +1,106 @@
...
+`cfgov.test.StdoutCapturingTestRunner` and can be enabled with the `TEST_RUNNER`
+environment variable:
+
+```sh
+TEST_RUNNER=cfgov.test.StdoutCapturingTestRunner tox -e fast
 ```
 
-To run both `flake8` and `isort` with Python 3, run
+This test runner is enabled when tests are run automatically on [Travis CI](https://travis-ci.org/),
+but is not used by defalt when running tests locally.
```
Comment (chosak) [inline docs/testing-be.md]: Typo: "defalt".

Comment (willbarton) [reply docs/testing-be.md]: Fixed in de901a8.

**Pull request merged**